PostHog/PostHogContext.swift

```diff
@@ -53,30 +53,73 @@ class PostHogContext {
             properties["$is_emulator"] = false
         #endif
 
+        // iOS app running in compatibility mode (Designed for iPad/iPhone)
+        var isiOSAppOnMac = false
+        #if os(iOS)
+            if #available(iOS 14.0, *) {
+                isiOSAppOnMac = ProcessInfo.processInfo.isiOSAppOnMac
+            }
+        #endif
+
+        // iPad app running on Mac Catalyst
+        #if targetEnvironment(macCatalyst)
+            let isMacCatalystApp = true
+        #else
+            let isMacCatalystApp = false
+        #endif
+
+        properties["$is_ios_running_on_mac"] = isiOSAppOnMac
+        properties["$is_mac_catalyst_app"] = isMacCatalystApp
+
         #if os(iOS) || os(tvOS)
             let device = UIDevice.current
             // use https://github.com/devicekit/DeviceKit
-            properties["$device_name"] = device.model
-            properties["$os_name"] = device.systemName
-            properties["$os_version"] = device.systemVersion
-
-            var deviceType: String?
-            switch device.userInterfaceIdiom {
-            case UIUserInterfaceIdiom.phone:
-                deviceType = "Mobile"
-            case UIUserInterfaceIdiom.pad:
-                deviceType = "Tablet"
-            case UIUserInterfaceIdiom.tv:
-                deviceType = "TV"
-            case UIUserInterfaceIdiom.carPlay:
-                deviceType = "CarPlay"
-            case UIUserInterfaceIdiom.mac:
-                deviceType = "Desktop"
-            default:
-                deviceType = nil
-            }
-            if deviceType != nil {
-                properties["$device_type"] = deviceType
+            let processInfo = ProcessInfo.processInfo
+
+            if isMacCatalystApp || isiOSAppOnMac {
+                let underlyingOS = device.systemName
+                let underlyingOSVersion = device.systemVersion
+                let macOSVersion = processInfo.operatingSystemVersionString
+
+                if isMacCatalystApp {
+                    let processInfo = ProcessInfo.processInfo
+                    properties["$os_name"] = "macOS \(processInfo.operatingSystemVersionString)" // eg Version 14.2.1 (Build 23C71)
```

Comment:
this should be os_name and not version

Reply:
mm i feel like the naming should be deterministic since it can be used for filtering.
having the version as part of it feels wrong.
we have os_version for that.
I'd remove from both.